src/main/java/com/ays/backend/user/service/impl/AuthServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.ays.backend.user.service.impl;
+
+import com.ays.backend.user.controller.payload.request.RegisterRequest;
+import com.ays.backend.user.controller.payload.response.AuthResponse;
+import com.ays.backend.user.model.entities.Organization;
+import com.ays.backend.user.model.entities.User;
+import com.ays.backend.user.model.enums.UserRole;
+import com.ays.backend.user.repository.UserRepository;
+import com.ays.backend.user.security.JwtTokenProvider;
+import com.ays.backend.user.service.AuthService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+
+@Service
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class AuthServiceImpl implements AuthService {
```

Comment:
We can make the implementation classes package-private so that they can only be accessed through interfaces.